cluster/swarm/cluster.go

```diff
@@ -255,7 +262,7 @@ func (c *Cluster) listNodes() []*node.Node {
 	c.RLock()
 	defer c.RUnlock()
 
-	out := []*node.Node{}
+	out := make([]*node.Node, 0, len(c.engines))
```

Comment:
There's no point  except a performance. If the language provides a mechanism to save memory allocations, it could be used to speedup our code. Also we may avoid creating useless pointers.
I hope go compiler will become smart enough to optimize such cases by itself.

Anyway if it's a code style thing, I can agree.